instrumentation/net/http/otelhttp/client_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelhttp
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"go.opentelemetry.io/otel/api/global"
+
+	mocktrace "go.opentelemetry.io/contrib/internal/trace"
+)
+
+func TestConvenienceWrappers(t *testing.T) {
+	provider, tracer := mocktrace.NewTracerProviderAndTracer(instrumentationName)
+	global.SetTracerProvider(provider)
+
+	content := []byte("Hello, world!")
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("traceparent") == "" {
+			t.Fatal("Expected traceparent header")
+		}
+		if _, err := w.Write(content); err != nil {
+			t.Fatal(err)
+		}
+	}))
+	defer ts.Close()
+
+	context, span := tracer.Start(context.Background(), "parent")
+	defer span.End()
+
+	res, err := Get(context, ts.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	res.Body.Close()
+
+	res, err = Head(context, ts.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	res.Body.Close()
+
+	res, err = Post(context, ts.URL, "text/plain", strings.NewReader("test"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	res.Body.Close()
+
+	form := make(url.Values)
+	form.Set("foo", "bar")
+	res, err = PostForm(context, ts.URL, form)
+	if err != nil {
+		t.Fatal(err)
+	}
+	res.Body.Close()
+
+	if len(tracer.EndedSpans()) != 4 {
+		t.Fatal("expected 4 client spans")
```

Comment:
I'm not sure that these tests are effective.  While they are now asserting that four spans were indeed created, it would be possible for them to pass if the `Get` method invoked itself four times and the other three were noops.  Perhaps this can be broken out into individual test cases for each method that can give higher confidence that the correct behavior is obtained?

Reply:
I added a check that we got the expected ([GET,HEAD,POST,POST]) span names.